mrp_multi_level/models/product_mrp_area.py

```diff
@@ -191,3 +196,8 @@ def _adjust_qty_to_order(self, qty_to_order):
         if self.mrp_maximum_order_qty and qty_to_order > self.mrp_maximum_order_qty:
             return self.mrp_maximum_order_qty
         return qty_to_order
+
+    def update_min_qty_from_main_supplier(self):
+        self.ensure_one()
```

Comment:
why? you can make it work for a recorset with a simple loop, this way it could be used directly in a server action.

Reply:
OK! method modified to fit server actions.